byway/src/main/webapp/destinations.js

```diff
@@ -224,15 +236,52 @@ window.onload = function(){
   });
   document.getElementById('user-input-form').addEventListener('submit', (event) => {
     event.preventDefault();
-    const formData = new  FormData(document.getElementById("user-input-form"));
-    fetch(buildUrlWithParams("/api/destinations", {tripKey}), {method: 'POST', body:formData}).then((response)=>
+    getDestinationPlaceId();
+    getStartPlaceId();
+  });
+}
+
+function getStartPlaceId(){
+  let formData = new FormData();
+  let service = new google.maps.places.PlacesService(map);
+  const startRequest = {
+    query: String(document.getElementById("start-search-box").value),
+    fields: ["place_id"]
+  };
+  service.findPlaceFromQuery(startRequest, (results, status) => {
+    if (status === google.maps.places.PlacesServiceStatus.OK) {
+      formData.append("start-location", results[0].place_id);
+      fetch(buildUrlWithParams("/api/destinations", {tripKey}), {method: 'POST', body:formData}).then((response)=>
         response.json()).then(locationData => {
-        updateLocations(locationData);
         updateStartDestination(locationData);
-      });
-    });
+      }); 
+    }
+    else{
+      alert("Location Invalid");
+    }
+  });
 }
 
+function getDestinationPlaceId(){
+  let formData = new FormData();
+  let service = new google.maps.places.PlacesService(map);
+  const destinationRequest = {
+    query: String(document.getElementById("destinations-search-box").value),
+    fields: ["place_id"]
+  };
+  service.findPlaceFromQuery(destinationRequest, (results, status) => {
+    if (status === google.maps.places.PlacesServiceStatus.OK) {
+      formData.append("destinations", results[0].place_id);
+      fetch(buildUrlWithParams("/api/destinations", {tripKey}), {method: 'POST', body:formData}).then((response)=>
+        response.json()).then(locationData => {
+        updateLocations(locationData);
+      }); 
```

Comment:
nit: align and format code by having the fetch in one line, and separating following promises per line, ie

```
fetch(...)
.then(...)
.then(...)
```

Reply:
done